gateways/istio-egress/templates/autoscale.yaml

```diff
@@ -10,7 +10,15 @@ metadata:
     release: {{ .Release.Name }}
 spec:
   maxReplicas: {{ $gateway.autoscaleMax }}
+{{- if $gateway.autoscaleMin }}
```

Comment:
Seem still an issue here, what if autoscaleMin is 1 and the actual replica is 1, in that case, the default PDB(with minAvailable=1 ) still prevents node drain.

Reply:
Can we just document that if pod disruption budget is set, min must be set to at least 2 ?
It seems like a lot of complexity - normal k8s configs just work.

Keep in mind we want to start using kustomize as well - where we don't have all the template magic.